File: memory.py
import os
import psutil
import decimal
from random import randrange
import math

total = 0
process = psutil.Process(os.getpid())
one_place = decimal.Decimal(10) ** -1

print("VANILLA\n")

used = 0
for i in range(15):
    heap = process.memory_info().rss
    used += heap
    total += used
    print("Iteration " + str(i + 1) + ": " + str(decimal.Decimal(used / 1e+6).quantize(one_place)) + " MB")

print("\n\nLOGIC STATEMENTS\n")

used = 0
for i in range(15):
    a = randrange(11)
    b = randrange(11)
    c = randrange(11)

    if a is b and a is c or a is not b or a is not c and a is b:
        heap = process.memory_info().rss
    else:
        heap = process.memory_info().rss

    used += heap
    total += used
    print("Iteration " + str(i + 1) + ": " + str(decimal.Decimal(used / 1e+6).quantize(one_place)) + " MB")

print("\n\nARITHMETIC OPERATIONS\n")

used = 0
for i in range(15):
    a = (((math.floor(randrange(11)) + 3) * 8) / 4) - (6 % math.sin(math.pi / 23))
    b = math.cos((a * -1) / math.sin(5 + 2))
    c = math.sqrt(((b % 3) + (a + 5) - b)) / math.pow(4, 3 * (2 * 9))

    heap = process.memory_info().rss
    used += heap
    total += used
    print("Iteration " + str(i + 1) + ": " + str(decimal.Decimal(used / 1e+6).quantize(one_place)) + " MB")

print("\n\nTOTAL USAGE: " + str(decimal.Decimal(total / 1e+6).quantize(one_place)) + " MB")

File: memory.js
var used, i, heap, a, b, c;
var total = 0;

console.log("VANILLA\n");

used = 0;
for (i = 0; i < 15; i++) {
    heap = process.memoryUsage();
    used += heap.heapUsed;
    total += used;
    console.log("Iteration " + (i + 1) + ": " + (used / 1e+6).toFixed(1) + " MB");
}

console.log("\n\nLOGIC STATEMENTS\n");

used = 0;
for (i = 0; i < 15; i++) {
    a = Math.floor(Math.random() * 10);
    b = Math.floor(Math.random() * 10);
    c = Math.floor(Math.random() * 10);

    if (a == b && a == c || a != b || a != c && a == b) {
        heap = process.memoryUsage();
    } else {
        heap = process.memoryUsage();
    }

    used += heap.heapUsed;
    total += used;
    console.log("Iteration " + (i + 1) + ": " + (used / 1e+6).toFixed(1) + " MB");
}

console.log("\n\nARITHMETIC OPERATIONS\n");

used = 0;
for (i = 0; i < 15; i++) {
    a = (((Math.floor(Math.random() * 10) + 3) * 8) / 4) - (6 % Math.sin(Math.PI / 23));
    b = Math.cos((a * -1) / Math.sin(5 + 2));
    c = Math.sqrt(((b % 3) + (a + 5) - b)) / Math.pow(4, 3 * (2 * 9));

    heap = process.memoryUsage();
    used += heap.heapUsed;
    total += used;
    console.log("Iteration " + (i + 1) + ": " + (used / 1e+6).toFixed(1) + " MB");
}

console.log("\n\nTOTAL USAGE: " + (total / 1e+6).toFixed(1) + " MB");